### ISSUE TITLE
NullReferenceException in tests/src/GC/API/GC/GetTotalAllocatedBytes.cs

### ISSUE THREAD

**Issue by kouvel** [dotnet/coreclr#24580]
See https://dev.azure.com/dnceng/public/_build/results?buildId=186486&view=ms.vss-test-web.build-test-results-tab

```
Unhandled Exception: System.TypeInitializationException: The type initializer for 'Test' threw an exception. ---> System.NullReferenceException: Object reference not set to an instance of an object.
at Test.Get_GetTotalAllocatedBytesDelegate() in /__w/1/s/tests/src/GC/API/GC/GetTotalAllocatedBytes.cs:line 26
at Test..cctor() in /__w/1/s/tests/src/GC/API/GC/GetTotalAllocatedBytes.cs:line 15
--- End of inner exception stack trace ---
at Test.TestSingleThreaded() in /__w/1/s/tests/src/GC/API/GC/GetTotalAllocatedBytes.cs:line 70
at Test.Main() in /__w/1/s/tests/src/GC/API/GC/GetTotalAllocatedBytes.cs:line 161
/root/helix/work/workitem/GC/API/GC/GetTotalAllocatedBytes/GetTotalAllocatedBytes.sh: line 259: 589 Aborted (core dumped) $LAUNCHER $ExePath \"${CLRTestExecutionArguments[@]}\"

Return code: 1
Raw output file: /root/helix/work/workitem/GC/API/Reports/GC.API/GC/GetTotalAllocatedBytes/GetTotalAllocatedBytes.output.txt
Raw output:
BEGIN EXECUTION
/root/helix/work/correlation/corerun GetTotalAllocatedBytes.exe ''
Expected: 100
Actual: 134
END EXECUTION - FAILED
Test Harness Exitcode is : 1
To run the test:
> set CORE_ROOT=/root/helix/work/correlation
> /root/helix/work/workitem/GC/API/GC/GetTotalAllocatedBytes/GetTotalAllocatedBytes.sh
Expected: True
Actual: False
```

**Comment by VSadov** [comment]
This kind of error could happen if corlib does not have `GC.GetTotalAllocatedBytes` method.

We had once an issue with the method being removed by `ILTrim`, but that was dealt with by adding the method to the exclusion list.

**Comment by echesakovMSFT** [comment]
The issue with https://dev.azure.com/dnceng/public/_build/results?buildId=186486&view=ms.vss-test-web.build-test-results-tab is that Build and Tests jobs are based on different HEADs of master branch.

If you look at **Build Linux arm64 checked** - Checkout step
```
HEAD is now at 8db2444... Merge 576983160168cca5fa45380f454eb41462567c5d into 15c4e02037e535a898ddd24390894adc8f3b5b06
```

and at **Test Pri0 Linux arm64_checked** - Checkout step
```
HEAD is now at 3ee0667... Merge 576983160168cca5fa45380f454eb41462567c5d into c4a0253f523570e83a547df8700e94c3647fdad0
```
The failing test was added in 934d73fea95b4c479b67fee0ff53caea4a325ee5 which is in between of 15c4e02037e535a898ddd24390894adc8f3b5b06
 and c4a0253f523570e83a547df8700e94c3647fdad0. So it's not a product issue - it's a bug with Checkout step in Azure Pipelines which we have already reported.

I have seen exactly the same issue in one of my PRs and clicking "Retry" button doesn't really help since it only retries the failing jobs (i.e. Test jobs) and the Build jobs are stuck with old product artifacts. The "workaround" for issues like this is to click "Rebuild" button instead of "Retry".

/cc @dotnet/jit-contrib In case if someone hits similar problem in the future.
